lib/src/database/matrix_sdk_database.dart

```diff
@@ -338,7 +338,32 @@ class MatrixSdkDatabase extends DatabaseApi with DatabaseFileStorage {
   }
 
   @override
-  Future<void> clear() => _collection.clear();
+  Future<void> clear() async {
+    _clientBox.clearCache();
+    _accountDataBox.clearCache();
+    _roomsBox.clearCache();
+    _preloadRoomStateBox.clearCache();
+    _nonPreloadRoomStateBox.clearCache();
+    _roomMembersBox.clearCache();
+    _toDeviceQueueBox.clearCache();
+    _roomAccountDataBox.clearCache();
+    _inboundGroupSessionsBox.clearCache();
+    _inboundGroupSessionsUploadQueueBox.clearCache();
+    _outboundGroupSessionsBox.clearCache();
+    _olmSessionsBox.clearCache();
+    _userDeviceKeysBox.clearCache();
+    _userDeviceKeysOutdatedBox.clearCache();
+    _userCrossSigningKeysBox.clearCache();
+    _ssssCacheBox.clearCache();
+    _presencesBox.clearCache();
+    _timelineFragmentsBox.clearCache();
+    _eventsBox.clearCache();
+    _seenDeviceIdsBox.clearCache();
+    _seenDeviceKeysBox.clearCache();
+    _userProfilesBox.clearCache();
```

Comment:
Can we move this into the Collection implementation? There we have a list of names `BoxCollection.boxNames` which we could use to iterate over them and then we cannot forget a box here

Reply:
I tried to do it but we create the Boxes here, so using the boxNames we can only create a new object that doesn't have this cache

Reply:
Do you think it's okay to do even more refactoring here?

Reply:
This is similar to what clearCache does below, so I think this is probably fine

Reply:
Overall I would like a refactor where we create, clear and delete boxes from a list of boxnames and not have to spell them out everytime, but we can look into that later ig